src/util/request.rs

```diff
@@ -0,0 +1,56 @@
+use reqwest::{
+    header::{COOKIE, USER_AGENT},
+    IntoUrl, Response,
+};
+use serde::Serialize;
+
+use crate::error::AocError;
+
+pub struct AocRequest
+{
+    client: reqwest::Client,
+}
+
+impl AocRequest
+{
+    const AOC_USER_AGENT: &str = "https://github.com/seblj/cargo-aoc by seblyng98@gmail.com";
```

Comment:
More nitpicking! Maybe `https://` is redundant?

Reply:
Agreed!